From 7eda133387425c0b910b0649e9cd6513bfe6246f Mon Sep 17 00:00:00 2001
From: Andrew Burgess <andrew.burgess@embecosm.com>
Date: Tue, 6 Mar 2018 13:05:34 +0000
Subject: [PATCH 3/9] gdb/riscv: Additional print format string fixes

Another riscv format string fix.

gdb/ChangeLog:

	* riscv-tdep.c (riscv_push_dummy_call): Use core_addr_to_string_nz
	where appropriate.
---
 gdb/ChangeLog    | 5 +++++
 gdb/riscv-tdep.c | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/gdb/ChangeLog b/gdb/ChangeLog
index 899cdcdeac..4006dcbdf3 100644
--- a/gdb/ChangeLog
+++ b/gdb/ChangeLog
@@ -1,5 +1,10 @@
 2018-08-23  Andrew Burgess  <andrew.burgess@embecosm.com>
 
+	* riscv-tdep.c (riscv_push_dummy_call): Use core_addr_to_string_nz
+	where appropriate.
+
+2018-03-06  Andrew Burgess  <andrew.burgess@embecosm.com>
+
 	* riscv-tdep.c (riscv_print_arg_location): Add header comment,
 	change parameter type.  Use GDB's print functions, and use
 	core_addr_to_string where appropriate.
diff --git a/gdb/riscv-tdep.c b/gdb/riscv-tdep.c
index 5c04a740df..797ca72a84 100644
--- a/gdb/riscv-tdep.c
+++ b/gdb/riscv-tdep.c
@@ -2121,8 +2121,8 @@ riscv_push_dummy_call (struct gdbarch *gdbarch,
 			      call_info.memory.arg_offset);
 	  fprintf_unfiltered (gdb_stdlog, "Stack required (for refs): 0x%x\n",
 			      call_info.memory.ref_offset);
-	  fprintf_unfiltered (gdb_stdlog, "          Stack allocated: 0x%lx\n",
-			      (osp - sp));
+	  fprintf_unfiltered (gdb_stdlog, "          Stack allocated: %s\n",
+			      core_addr_to_string_nz (osp - sp));
 	}
     }
 
-- 
2.18.0

